dotnet-engine/Yggdrasil.Engine/NativeLoader.cs

```diff
@@ -0,0 +1,179 @@
+                    {
+                        FileName = "ldd",
+                        Arguments = "--version",
+                        RedirectStandardOutput = true,
+                        UseShellExecute = false
+                    };
+                    process.Start();
+                    string lddOutput = process.StandardOutput.ReadToEnd();
+                    process.WaitForExit();
+
+                    return lddOutput.Contains("musl");
+                }
+            }
+            catch
+            {
+                return false;
+            }
+        }
+    }
+
+    internal static IntPtr LoadFunctionPointer(IntPtr libHandle, string functionName)
+    {
+        Type nativeLibraryType = Type.GetType("System.Runtime.InteropServices.NativeLibrary, System.Runtime.InteropServices");
+        if (nativeLibraryType != null)
+        {
+            var getExportMethod = nativeLibraryType.GetMethod("GetExport", new[] { typeof(IntPtr), typeof(string) });
+            if (getExportMethod != null)
+            {
+                return (IntPtr)getExportMethod.Invoke(null, new object[] { libHandle, functionName });
+            }
+        }
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            return GetProcAddress(libHandle, functionName);
+        else
+            return dlsym(libHandle, functionName);
+    }
+
+    private static IntPtr LoadBinary(string libPath)
+    {
+        IntPtr handle;
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            handle = LoadWindowsLibrary(libPath);
+        else
+            handle = LoadUnixLibrary(libPath);
+
+        if (handle == IntPtr.Zero)
+            throw new DllNotFoundException($"Failed to load library from {libPath}");
+
+        return handle;
+    }
+
+    private static IntPtr LoadUnixLibrary(string libPath)
+    {
+        // Try NativeLibrary.Load (works on .NET Core 3+)
+        Type nativeLibraryType = Type.GetType("System.Runtime.InteropServices.NativeLibrary, System.Runtime.InteropServices");
+        if (nativeLibraryType != null)
+        {
+            var loadMethod = nativeLibraryType.GetMethod("Load", new[] { typeof(string) });
+            if (loadMethod != null)
+                return (IntPtr)loadMethod.Invoke(null, new object[] { libPath });
+        }
+        return dlopen(libPath, RTLD_NOW);
+    }
+
+    private static IntPtr LoadWindowsLibrary(string libPath)
+    {
+        // Try NativeLibrary.Load (works on .NET Core 3+)
+        Type nativeLibraryType = Type.GetType("System.Runtime.InteropServices.NativeLibrary, System.Runtime.InteropServices");
+        if (nativeLibraryType != null)
+        {
+            var loadMethod = nativeLibraryType.GetMethod("Load", new[] { typeof(string) });
+            if (loadMethod != null)
+                return (IntPtr)loadMethod.Invoke(null, new object[] { libPath });
+        }
+
+        return LoadLibrary(libPath);
+    }
+
+#if NETSTANDARD
```

Comment:
Would any of this be easier if we used compiler directives and went back to outputting .NET Framework targets? (ie `net472` et al)

Reply:
Fair enough! Was worth asking the question though in case it could help reduce complexity